### PR TITLE
[Feature]  Modify guild features

### PR DESCRIPTION
**Pull request by Misha-133**
This PR adds support for modifying guild features & adds a few missing ones.

### Changes
- add missing guild features
- add `GuildFeatures` json writer
- add `GuildFeature` property to `GuildProperties`

### CODE REVIEW

**Review by csmir — CHANGES_REQUESTED** [review]
Would it not be wise to check the mutable features, to protect the developer from editing before unavailable ones get sent to discord? Maybe I read over this check, but I do not see it anywhere.

**Comment by Misha-133** [comment]
> Would it not be wise to check the mutable features, to protect the developer from editing before unavailable ones get sent to discord? Maybe I read over this check, but I do not see it anywhere.

Discord ignores immutable features
is this kind of validation needed in this case?

**Comment by csmir** [comment]
Just a comment now is fine then, I will merge it

File: src/Discord.Net.Core/Entities/Guilds/GuildProperties.cs [hunk 1]
```diff
@@ -113,5 +113,10 @@ public class GuildProperties
         ///     Gets or sets if the boost progress bar is enabled.
         /// </summary>
         public Optional<bool> IsBoostProgressBarEnabled { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the guild features enabled in this guild.
```
Comment (csmir) [inline src/Discord.Net.Core/Entities/Guilds/GuildProperties.cs]:
```suggestion
        ///     Gets or sets the guild features enabled in this guild. Features that are not mutable will be ignored.
```

**Review by csmir — DISMISSED** [review]

**Review by csmir — APPROVED** [review]

**Pull request merged**